### DISCUSSION TITLE
Bug? Report: Syntax error

### DISCUSSION THREAD

**Discussion by sour-dani** [sam-astro/Z-Sharp#36] [Q&A]
When defining a function not having space between the function parentheses and curly brackets causes a type error
```
func Main(){
    Printl("Howdy!")
}
```
```
ZSharp v2.1.2-alpha

WARNING: unrecognized type '}' on line: 2
```

Also will there be a syntax/intellisense plugin for VSC in the future?

**Comment by pmo6** [comment]
There is already a syntax highlighting feature in vscode. And I think that you may need to have the opening brace below the “func Main()” call, since that may have worked for me when I tried running it, but I find it weird that programming languages would even check which lines the braces were on. I’m sorry if I sound like I’m ranting, but I’d also like to bring up the fact that Go seems to force one to have the brace on the _same_ line as the function call.

**Reply by sour-dani** [reply]
My point is that I think it's weird that it has to have the space in-between the parentheses and curly brackets.

As for the syntax highlighting is there a way to enable that? Just opening the file as is in VSC doesn't enable any highlighting for me.

**Reply by pmo6** [reply]
You need to download the ZSharp plugin first.

**Reply by sour-dani** [reply]
huh, somehow i missed it when i was first trying this out. 

thanks again!

**Reply by binyamin555** [reply]
This is probably due to how the parser is written and is probably not meant to be part of the language

**Reply by sam-astro** [reply]
Definitely, I built the parser in a very poor way, which causes many bad things to happen, such as simple syntax causing errors etc. To fix these issues I think a total rewrite would be required, (among fixing other issues), though I'm not working on this anymore so it doesn't matter too much.

**Reply by binyamin555** [reply]
Yeah, so I looked at the parser to see how hard it'll be rewriting it and, as you said, found it'll need an entire rewrite with proper planning.

I don't even know how you managed to get this work without an AST.
